jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -131,15 +131,8 @@
 	static final Unsafe getUnsafe() {
 		return UnsafeGetter.myUnsafe;
 	}
-	
-	private static final class JITHelpersGetter {
-		public static final JITHelpers myJITHelpers = JITHelpers.getHelpers();
-	}
+	static final JITHelpers myJITHelpers = JITHelpers.getHelpers();
```

Comment:
I assume this was done to avoid initializing `myJITHelpers` as part of MethodHandle class initialization, and instead initialized it when needed. I checked where it's called from, and it appears that in most cases, the interpreter won't use it.

Although this might have a minor startup impact, I don't think the class initialization of JITHelpers is very expensive, so it might not be worth the extra class. I just wanted to point out the likely reason for this approach (for the record).

Reply:
Thanks for giving a thought to the rationale behind the getter. It's possible that the getter used to helpe the startup before JITHelpers is used so intensively. Currently, it is used everywhere in internal code, for example in String, so it will be initialized very early and calling its method in other classes' initializer should be fairly cheap nowadays.